src/tiramisu_function.cpp

```diff
@@ -116,6 +116,112 @@ isl_union_map *tiramisu::function::compute_dep_graph() {
     return result;
 }
 
+const std::map<std::string, tiramisu::buffer *> tiramisu::function::get_mapping() const
+{
+  return this->mapping;
+
+}
+
+void  tiramisu::function::add_mapping(std::pair<std::string,tiramisu::buffer *> p)
+{
+  this->mapping.insert(p);
+
+}
+
+	
+const int  &function::Automatic_communication(tiramisu::computation* c1,tiramisu::computation* c2 ) const
+{
+    std::map<std::string, tiramisu::buffer*> buff = this->get_buffers();
+    std::map<std::string, tiramisu::buffer*> mp = this->mapping;
+    std::map<std::string, tiramisu::buffer*>::iterator it ;
+    std::string name, cpt_name;
+    int i = 1;
+    char tag;  
+    tiramisu::computation* first_cpt = c1;
+    tiramisu::computation* last_cpt = c2;
+    for (it = buff.begin(); it != buff.end(); ++it)
+    {
+      name = it->second->get_name();
+      tag = it->second->tag;
+      if (it->second->get_argument_type() != tiramisu::a_temporary)
+        {
+            cpt_name= "cpt" + std::to_string(i); 
+            i++;
+            switch (tag)
+            {
+                case 'c':
+                if ((it->second->get_argument_type() == tiramisu::a_input) && (mp.find(name) != mp.end()))
+                {
+                   if (buff.find(mp.find(name)->second->get_name())->second->auto_trans== true) 
```

Comment:
This code is too complicated. Can you put "buff.find(mp.find(name)->second->get_name())" in a variable then use that variable?